src/applications/vaos/reducers/newAppointment.js

```diff
@@ -510,12 +513,40 @@ export default function formReducer(state = initialState, action) {
         appointmentLength: action.appointmentLength,
       };
     }
-    case FORM_FETCH_AVAILABLE_APPOINTMENTS_FAILED: {
+    case FORM_CALENDAR_FETCH_SLOTS_FAILED: {
       return {
         ...state,
         appointmentSlotsStatus: FETCH_STATUS.failed,
       };
     }
+    case FORM_CALENDAR_ON_CHANGE: {
+      return {
+        ...state,
+        data: {
+          ...state.data,
+          calendarData: action.calendarData,
+        },
+      };
+    }
+    case FORM_CALENDAR_VALIDATE: {
+      return {
+        ...state,
+        data: {
+          ...state.data,
+          calendarData: { ...state.data.calendarData, error: action.error },
+        },
+      };
+    }
+    case FORM_CALENDAR_CLEAR_DATA: {
```

Comment:
If we move the validation out of Redux, do we still need this action? I think, any state other than the selected timeslots, that we're just going to throw away on a page change, could stay out of Redux.

Reply:
Removed